src/commands/deploy.rs

```diff
@@ -163,6 +194,31 @@ impl DeployCommand {
         Ok(())
     }
 
+    async fn compute_buildinfo(&self, cfg: &RawAppManifest) -> Result<BuildMetadata> {
+        let mut sha256 = Sha256::new();
+
+        for x in cfg.components.iter() {
+            match &x.source {
+                config::RawModuleSource::FileReference(p) => {
+                    let mut r = File::open(p)?;
+                    let _ = copy(&mut r, &mut sha256)?;
+                }
+                config::RawModuleSource::Bindle(_b) => {}
+            }
+        }
+
+        let mut r = File::open(&self.app)?;
+        let _ = copy(&mut r, &mut sha256)?;
+
+        let mut final_digest = format!("q{:x}", sha256.finalize());
```

Comment:
I think this could produce a repeat result if the `spin.toml` and the modules were all the same, but the set of assets had changed e.g. a component still has the same Wasm module and still has `*.jpg` for its files but the user has created a new `.jpg` file in the source directory...?

Reply:
We addressed this by using `assets::collect` to write the digests of component assets into the build info. Verified using the local installer.

Reply:
The bindle publisher already collects and hashes all the assets so we could do this as part of that pass rather than revisiting all the files again.  But we can look at whether that's worth doing after the 0.3 crunch. - it's definitely not needed.